src/main/java/org/folio/service/finance/budget/BudgetRestrictionService.java

```diff
@@ -31,20 +33,27 @@ public void checkEncumbranceRestrictions(List<? extends EncumbranceRelationsHold
         .filter(EncumbranceRelationsHolder::getRestrictEncumbrance)
         .collect(groupingBy(EncumbranceRelationsHolder::getBudget));
 
-    List<String> failedBudgetIds = budgetHoldersMap.entrySet()
+    Map<String, String> fundHoldersMap = dataHolders.stream()
+        .filter(EncumbranceRelationsHolder::getRestrictEncumbrance)
+        .map(EncumbranceRelationsHolder::getFundDistribution)
+        .filter(fd -> fd != null && fd.getFundId() != null && fd.getCode() != null)
+        .collect(Collectors.toMap(FundDistribution::getFundId, FundDistribution::getCode,
```

Comment:
Please add before collect distinct if possible to have fundDistributions with same fundId

Reply:
Distinct added

Reply:
I think adding distinct was not needed, because there was already a merge function for toMap to deal with duplicate keys.